ocf_datapipes/load/gsp.py

```diff
@@ -0,0 +1,120 @@
+import datetime
+from typing import Optional
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import xarray as xr
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+
+try:
+    from ocf_datapipes.utils.eso import get_gsp_metadata_from_eso, get_gsp_shape_from_eso
+
+    _has_pvlive = True
+except ImportError:
+    print("Unable to import PVLive utils, please provide filenames with OpenGSP")
+    _has_pvlive = False
+
+
+@functional_datapipe("open_gsp")
+class OpenGSPIterDataPipe(IterDataPipe):
+    def __init__(
+        self,
+        gsp_pv_power_zarr_path: str,
+        gsp_id_to_region_id_filename: Optional[str] = None,
+        sheffield_solar_region_path: Optional[str] = None,
```

Comment:
could rename to gsp_regions_path, as NG_ESO publish them. I know solar sheffield make them, but make we get them from NG.

Reply:
Okay, yeah, sounds good!